castai/resource_castai_cluster.go

```diff
@@ -172,9 +309,54 @@ func resourceCastaiClusterRead(ctx context.Context, data *schema.ResourceData, m
 		data.Set(ClusterFieldKubeconfig, []interface{}{})
 	}
 
+	policies, err := client.GetPoliciesWithResponse(ctx,sdk.ClusterId(data.Id()))
+	if checkErr := sdk.CheckGetResponse(policies, err); checkErr == nil {
+		if err := data.Set(PolicyFieldAutoscalerPolicies, flattenAutoscalerPolicies(policies.JSON200)); err != nil {
+			return nil
+		}
+	} else {
+		log.Printf("[WARN] autoscaling policies are not available for cluster %q: %v", data.Id(), checkErr)
+	}
+
 	return nil
 }
 
+
+
+func resourceCastaiClusterUpdate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfig).api
+
+	log.Printf("[DEBUG] Cluster %q autoscaling policies update", data.Id())
+	autoscalerParams, ok := data.Get(PolicyFieldAutoscalerPolicies).([]interface{})
+	if !ok || autoscalerParams == nil || len(autoscalerParams) == 0 || autoscalerParams[0] == nil {
+		log.Printf("[DEBUG] Reading Policies `autoscaler_policies` empty parameters %v")
+		return resourceCastaiClusterRead(ctx, data, meta)
+	}
+	updatePolicies(ctx, client, data.Id(), autoscalerParams[0].(map[string]interface{}))
```

Comment:
The description and title mentions no update on creation, but there is update logic.

Reply:
As our CreateCluster mothership API does not allow to set policies on cluster creation, we need to update settings in same Create phase. Adjusting log statement.